cogito/putter.go

```diff
@@ -134,22 +135,40 @@ func (putter *ProdPutter) ProcessInputDir() error {
 	return nil
 }
 
-func (putter *ProdPutter) Sinks() []Sinker {
-	return []Sinker{
-		GitHubCommitStatusSink{
+func (putter *ProdPutter) Sinks() ([]Sinker, error) {
+	var err error
+	supportedSinks := map[string]Sinker{
+		"github": GitHubCommitStatusSink{
 			Log:     putter.log.Named("ghCommitStatus"),
 			GhAPI:   putter.ghAPI,
 			GitRef:  putter.gitRef,
 			Request: putter.Request,
 		},
-		GoogleChatSink{
+		"gchat": GoogleChatSink{
 			Log: putter.log.Named("gChat"),
 			// TODO putter.InputDir itself should be of type fs.FS.
 			InputDir: os.DirFS(putter.InputDir),
 			GitRef:   putter.gitRef,
 			Request:  putter.Request,
 		},
 	}
+	sinksParams := putter.Request.Params.Sinks
+	if len(sinksParams) == 0 {
+		// No sink specified, it will return all supported sinks.
+		sinksParams = make([]string, 0, len(supportedSinks))
+		for k := range supportedSinks {
+			sinksParams = append(sinksParams, k)
+		}
+	}
+	sinkers := make([]Sinker, 0, len(sinksParams))
+	for _, s := range sinksParams {
+		sinkers = append(sinkers, supportedSinks[s])
```

Comment:
What will happen if `s` is not in `supportedSinks`?

Reply:
It's tested here: https://github.com/Pix4D/cogito/pull/109/files#diff-4642f345de08ea53e3e6fe15f6b457e7052f316cda64aca54e21f78f51204cb3R203  (the other PR). The  cinfured sink is `pizza` and the error `put: source: invalid sink: [pizza]. Supported sinks: 'gchat', 'github'`